File: selenium-agent-mcp-starter-python/src/pages/login_page.py
from selenium.webdriver.common.by import By
from src.pages.base_page import BasePage


class LoginPage(BasePage):
    _login_logo = (By.CSS_SELECTOR, ".login_logo")
    _username_input = (By.ID, "user-name")
    _password_input = (By.ID, "password")
    _login_button = (By.ID, "login-button")
    _error_message = (By.CSS_SELECTOR, "h3[data-test='error']")

    def is_loaded(self) -> bool:
        return self.is_visible(self._login_logo)

    def login(self, username: str, password: str):
        self.type(self._username_input, username)
        self.type(self._password_input, password)
        self.click(self._login_button)

    def get_error_message(self) -> str:
        return self.get_text(self._error_message)

File: selenium-agent-mcp-starter-java/src/main/java/com/example/pages/LoginPage.java
package com.example.pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

public class LoginPage extends BasePage {
    private By usernameInput = By.id("username");
    private By passwordInput = By.id("password");
    private By loginButton = By.cssSelector("button[type='submit']");
    private By errorMessage = By.cssSelector(".error");

    public LoginPage(WebDriver driver) {
        super(driver);
    }

    public void login(String username, String password) {
        type(usernameInput, username);
        type(passwordInput, password);
        click(loginButton);
    }

    public String getErrorMessage() {
        return getText(errorMessage);
    }
}